backend/src/main/java/fk/prof/backend/leader/election/LeaderElectedTask.java

```diff
@@ -23,28 +36,70 @@ private LeaderElectedTask(Vertx vertx,
 
     if (backendDeployments != null) {
       // NOTE: If backend deployments supplied, leader will not serve as backend and only do leader related operations
-      this.backendVerticlesUndeployer = () -> backendDeployments.stream().forEach(deploymentId ->
+      this.backendVerticlesUndeployer = () -> {
+        logger.info("Unloading backend verticles");
+        List<Future> undeployFutures =  new ArrayList<>();
+        for(String deploymentId: backendDeployments) {
+          Future<Void> future = Future.future();
+          undeployFutures.add(future);
+
           vertx.undeploy(deploymentId, result -> {
             if (result.succeeded()) {
-              if(logger.isDebugEnabled()) {
-                logger.debug(String.format("Successfully un-deployed backend verticle=%s", deploymentId));
-              }
+              logger.info(String.format("Successfully un-deployed backend verticle=%s", deploymentId));
+              future.complete();
             } else {
               logger.error(String.format("Error when un-deploying backend verticle=%s", deploymentId), result.cause());
+              future.fail(result.cause());
             }
-          })
-      );
+          });
+        }
+        return CompositeFuture.all(undeployFutures);
+      };
     }
 
     this.leaderHttpVerticlesDeployer = Preconditions.checkNotNull(leaderHttpVerticleDeployer);
   }
 
   @Override
   public void run() {
+    logger.info("Beginning leader elected task");
+    runTask().setHandler(ar -> {
+      if(ar.failed()) {
+        ctrFailure.inc();
```

Comment:
This counter will be useful for setting alert. Did you have anything else in mind?